src/pact.spec.ts

```diff
@@ -0,0 +1,263 @@
+			});
+		});
+
+		context("when user specifies invalid ssl", () => {
+			it("should return an error on non-boolean", () => {
+				expect(() => pact.createServer({ssl: 1} as any)).to.throw(Error);
+			});
+		});
+
+		context("when user specifies invalid cors", () => {
+			it("should return an error on non-boolean", () => {
+				expect(() => pact.createServer({cors: 1} as any)).to.throw(Error);
+			});
+		});
+
+		context("when user specifies invalid log", () => {
+			it("should return an error on invalid path", () => {
+				pact.createServer({log: path.resolve(dirPath, "log.txt")});
+				expect(fs.statSync(dirPath).isDirectory()).to.be.true;
+			});
+		});
+
+		context("when user specifies invalid spec", () => {
+			it("should return an error on non-number", () => {
+				expect(() => pact.createServer({spec: "1"} as any)).to.throw(Error);
+			});
+
+			it("should return an error on negative number", () => {
+				expect(() => {
+					pact.createServer({spec: -12});
+				}).to.throw(Error);
+			});
+
+			it("should return an error on non-integer", () => {
+				expect(() => {
+					pact.createServer({spec: 3.14});
+				}).to.throw(Error);
+			});
+		});
+
+		context("when user specifies invalid consumer name", () => {
+			it("should return an error on non-string", () => {
+				expect(() => pact.createServer({consumer: 1234} as any)).to.throw(Error);
+			});
+		});
+
+		context("when user specifies invalid provider name", () => {
+			it("should return an error on non-string", () => {
+				expect(() => pact.createServer({provider: 2341} as any)).to.throw(Error);
+			});
+		});
+	});
+
+	describe("List servers", () => {
+		context("when called and there are no servers", () => {
+			it("should return an empty list", () => {
+				expect(pact.listServers()).to.be.empty;
+			});
+		});
+
+		context("when called and there are servers in list", () => {
+			it("should return a list of all servers", () => {
+				pact.createServer({port: 1234});
+				pact.createServer({port: 1235});
+				pact.createServer({port: 1236});
+				expect(pact.listServers()).to.have.length(3);
+			});
+		});
+
+		context("when server is removed", () => {
+			it("should update the list", () => {
+				pact.createServer({port: 1234});
+				pact.createServer({port: 1235});
+				return pact.createServer({port: 1236})
+					.delete()
+					.then(() => expect(pact.listServers()).to.have.length(2));
+			});
+		});
+	});
+
+	describe("Remove all servers", () => {
+		context("when removeAll() is called and there are servers to remove", () => {
+			it("should remove all servers", () => {
+				pact.createServer({port: 1234});
+				pact.createServer({port: 1235});
+				pact.createServer({port: 1236});
+				return pact.removeAllServers()
+					.then(() => expect(pact.listServers()).to.be.empty);
+			});
+		});
+	});
+
+	// These tests never worked because the expect was wrong.  When fixed, massive issue ensues
+	xdescribe("Verify Pacts", () => {
```

Comment:
@mefellows these 2 tests had to be excluded as they simply wasn't working.  The expect wasn't doing it's job because it was doing `to.be.eventually.resolved;` which is not an expect type, hence it was ignoring the error.  I'm not sure what you were trying to test specifically since the pactUrls isn't valid and causing issues with the verifier.